### ISSUE TITLE
Catalog operator consumes 9GB of memory and shows errors of unhealty clients

### ISSUE THREAD

**Issue by ctron** [operator-framework/operator-lifecycle-manager#1015]
The `catalog-operator` pod builds up memory usage (9GB+):

~~~
$ kubectl top pod catalog-operator-64556ffff5-9lzpl
NAME                                CPU(cores)   MEMORY(bytes)   
catalog-operator-64556ffff5-9lzpl   18m          9636Mi
~~~

And complains in the log about un-healty clients:

~~~
…
time="2019-08-29T07:12:50Z" level=info msg="building connection to registry" currentSource="{olm-operators openshift-operator-lifecycle-manager}" id=5RB5r source=olm-operators
time="2019-08-29T07:12:50Z" level=info msg="client hasn't yet become healthy, attempt a health check" currentSource="{olm-operators openshift-operator-lifecycle-manager}" id=5RB5r source=olm-operators
time="2019-08-29T07:12:51Z" level=info msg="building connection to registry" currentSource="{certified-operators openshift-marketplace}" id=C8tH6 source=certified-operators
time="2019-08-29T07:12:51Z" level=info msg="client hasn't yet become healthy, attempt a health check" currentSource="{certified-operators openshift-marketplace}" id=C8tH6 source=certified-operators
time="2019-08-29T07:12:51Z" level=info msg="building connection to registry" currentSource="{community-operators openshift-marketplace}" id=uaPkG source=community-operators
time="2019-08-29T07:12:51Z" level=info msg="client hasn't yet become healthy, attempt a health check" currentSource="{community-operators openshift-marketplace}" id=uaPkG source=community-operators
time="2019-08-29T07:12:52Z" level=info msg="building connection to registry" currentSource="{redhat-operators openshift-marketplace}" id=l6uSz source=redhat-operators
time="2019-08-29T07:12:52Z" level=info msg="client hasn't yet become healthy, attempt a health check" currentSource="{redhat-operators openshift-marketplace}" id=l6uSz source=redhat-operators
…
~~~

**Comment by ecordell** [comment]
Thanks for the report - from the logs, I am assuming you are running under openshift 4.1.z?

If that's correct - we have this issue fixed in master and will backport shortly. If this is running a master of OLM then we'll want to dig a little more into this and gather more data.

**Comment by ctron** [comment]
> Thanks for the report - from the logs, I am assuming you are running under openshift 4.1.z?

Yes, that is correct.

> If that's correct - we have this issue fixed in master and will backport shortly.

I am looking forward to this! :-) thanks!

**Comment by ecordell** [comment]
The fix for this is backported to 4.1 now!
